app/models/static_content.rb

```diff
@@ -1,10 +1,12 @@
 # frozen_string_literal: true
 
 class StaticContent < ApplicationRecord
-  # we need the scope for destroying records, but we do not have any specifics
-  # for current users at static contents, so we return `all`
-  scope :filtered_for_current_user, lambda { |_current_user|
-    all
+  # we need the scope for destroying records, which only admins and editors are allowed to
+  scope :filtered_for_current_user, lambda { |current_user|
+    return all if current_user.admin_role?
+    return all if current_user.editor_role?
+
+    none
```

Comment:
I think this is fairly misleading

Reply:
why? we need an active record relation returned here, but without any data. what would be your different approach?

Reply:
Misleading, not wrong.
It doesn't "really" filter them for the `current_user`, I wrote an (in my opinion) better solution above.

Reply:
i understand. but we also need the scope for `Mutations::DestroyRecord` for example. so refactoring with your suggestion will lead to a more complex job across multiple places.